docs/src/main/sphinx/admin/properties-task.rst

```diff
@@ -125,15 +125,31 @@ the task has remaining splits to process.
 ^^^^^^^^^^^^^^^^^^^^^
 
 * **Type:** :ref:`prop-type-integer`
-* **Restrictions:** Must be a power of two
 * **Default value:** ``1``
 
-The number of concurrent writer threads per worker per query. Increasing this value may
-increase write speed, especially when a query is not I/O bound and can take advantage
-of additional CPU for parallel writes. Some connectors can be bottlenecked on CPU when
-writing due to compression or other factors. Setting this too high may cause the cluster
-to become overloaded due to excessive resource utilization. This can also be specified on
-a per-query basis using the ``task_writer_count`` session property.
+The number of concurrent writer threads per worker per query when
+:ref:`preferred partitioning <preferred-write-partitioning>` and
+:ref:`task writer scaling <prop-task-scale-writers>` are not used. Increasing this value may
+increase write speed, especially when a query is not I/O bound and can take advantage of
+additional CPU for parallel writes. Some connectors can be bottlenecked on CPU when writing
+due to compression or other factors. Setting this too high may cause the cluster to become
+overloaded due to excessive resource utilization. This can also be specified on a per-query
+basis using the ``task_writer_count`` session property.
+
+``task.partitioned-writer-count``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Restrictions:** Must be a power of two
+* **Default value:** ``32``
+
+The number of concurrent writer threads per worker per query when
+:ref:`preferred partitioning <preferred-write-partitioning>` is used. Increasing this value may
+increase write speed, especially when a query is not I/O bound and can take advantage of additional
+CPU for parallel writes. Some connectors can be bottlenecked on CPU when writing due to compression
+or other factors. Setting this too high may cause the cluster to become overloaded due to excessive
```

Comment:
nit: It would be good to somehow differentiate between `task.partitioned-writer-count` and `task.writer-count` in doc.

Setting `task.writer-count` high for partitioned insert will have much bigger impact on resource utilization that setting `task.partitioned-writer-count` high

Reply:
Added some more info in `task.writer-count` setting